biz.aQute.repository/src/aQute/p2/provider/ArtifactRepository.java

```diff
@@ -107,7 +107,8 @@ public Domain getParent() {
 
 		NodeList artifactNodes = getNodes("repository/artifacts/artifact");
 		for (int i = 0; i < artifactNodes.getLength(); i++) {
-			final XMLArtifact xmlArtifact = getFromType(artifactNodes.item(i), XMLArtifact.class);
+			final Node artifactNode = artifactNodes.item(i).cloneNode(true);
```

Comment:
It's a variant on a common XPath optimisation. Resolving relative paths is extremely slow normally as it resolves against the entire document. Cloning divorces the resulting node from its parent, so the resolve operates only on the relevant section. I think that's the gist of it anyway, and it certainly worked.